pl.c

```diff
@@ -459,6 +452,35 @@ static void pl_delete_selected_pl(void)
 	job_schedule_pl_delete(pdd);
 }
 
+static void pl_delete_selected_pl(void)
+{
+	if (list_len(&pl_head) == 1) {
+		error_msg("cannot delete the last playlist");
+		return;
+	}
+
+	if (yes_no_query("Delete selected playlist? [y/N]") != UI_QUERY_ANSWER_YES)
+		return;
+
+	pl_delete(pl_visible);
+}
+
+void pl_delete_all(void)
+{
+	struct playlist *pl;
+	struct playlist *temp;
+	struct playlist *last = NULL;
+	list_for_each_entry_safe(pl, temp, &pl_head, node) {
+		if (list_len(&pl_head) == 1) {
+			last = pl;
+			break;
+		}
+		pl_delete(pl);
+	}
+	pl_create_default();
```

Comment:
It would be cleaner to do this inside `pl_delete` where it checks if it's the last playlist. And at that point we could also remove the "cannot delete the last playlist" part of `pl_delete_selected_pl`.

Reply:
68cf608